README.md
<p align="center">
  <img src="https://cdn.jsdelivr.net/gh/PasteUs/CDN@0.0.12/screenshot/pasteme/duck.png" alt="" width=200>
</p>
<p align="center">
  <a href="./LICENSE">
    <img src="https://img.shields.io/eclipse-marketplace/l/notepad4e.svg" alt="">
  </a>
  <a>
    <img src="https://img.shields.io/badge/Frontend-3.2.2-brightgreen.svg" alt="version">
  </a>
  <a>
    <img src="https://img.shields.io/badge/GoBackend-3.2.1-lightblue.svg" alt="version">
  </a>
  <a>
    <img src="https://img.shields.io/badge/Admin-0.0.2-yellow.svg" alt="version">
  </a>
  <a href="#谢谢老板">
    <img src="https://img.shields.io/badge/%24-donate-ff69b4.svg" alt="donate">
  </a>
</p>
<div align="center">
  <h1>PasteMe</h1>
</div>

PasteMe 是一个无需注册的文本分享平台，针对代码提供了额外的高亮功能。

+ 在存储内容时，**设置密码**和**阅后即焚**可以高度保证用户内容的安全性和私密性。

+ 在将自己的内容分享给别人时，提供了一键复制链接和二维码分享等多种途径。

+ 在查看别人的内容时，可以一键复制所有文本。如果查看的是阅后即焚的内容，那么在网页加载完成之前，实体数据就已经不存在了。

各位老板 Star 、Issue 、PR 好不啦。**Orz**

## 一些场景

+ 如果你要发布一个脚本，可以把 `Bash` 或者 `Python` 等脚本上传至 `PasteMe` ，然后通过 `curl` 和管道机制来进行优雅的发布，比如：`curl api.pasteme.cn/8219 | python3`

+ 如果你要发给某人一些私密的信息，比如一段 `token`，但是通过 QQ 、微信等聊天工具可能会被 “查水表”，你可以将私密信息以阅后即焚形式上传至 `PasteMe` ，将一次性链接分享给别人，别人查看一次之后这个链接就会失效

+ 想要向服务器内粘贴一段代码，但是苦于字符集，复制、上传上去之后有其它的字符，此时你可以上传至 `pasteme` ，然后通过 `wget api.pasteme.cn/<key> -O file_name` 来进行优雅的拉取

+ 新装了一台没有图形化界面的服务器，没开 `sshd` 服务，没有可用的编辑器去编辑 `sources.list` 文件，直接用默认源安装一个编辑器又太慢，此时可以用 `curl api.pasteme.cn/<key> > /etc/apt/sources.list` 或 `wget api.pasteme.cn/<key> -O /etc/apt/sources.list` 来更新 `apt` 源，然后进行优雅的 `apt update`

+ 阅后即焚的链接是可以自定义的，比如 [pasteme.cn/example](https://pasteme.cn/example) ，更多详情请查看 [使用文档](./doc/DOCUMENT.md)

## 截图

<details>

![homePage](https://cdn.jsdelivr.net/gh/PasteUs/CDN@0.0.12/screenshot/pasteme/home.png)

![read_once](https://cdn.jsdelivr.net/gh/PasteUs/CDN@0.0.12/screenshot/pasteme/read_once.png)

![success](https://cdn.jsdelivr.net/gh/PasteUs/CDN@0.0.12/screenshot/pasteme/success.png)

![qrcode](https://cdn.jsdelivr.net/gh/PasteUs/CDN@0.0.12/screenshot/pasteme/qrcode.png)

![markdown_parsed](https://cdn.jsdelivr.net/gh/PasteUs/CDN@0.0.12/screenshot/pasteme/markdown_parsed.png)

![markdown_source](https://cdn.jsdelivr.net/gh/PasteUs/CDN@0.0.12/screenshot/pasteme/markdown_source.png)

</details>

## Deploy

[部署文档](./doc/DOCKER_DEPLOY.md)

### 更新日志

[PasteMe 更新日志](https://www.lucien.ink/pasteme_log.html)

### 捐助

#### 捐助名单

> 2020 年 5 月 1 日更新

小伙伴们在捐助的时候可以添加留言以告知自己的 `ID` ，如果是 `GitHub` 账号的话我会顺便 `@` 出来。

> 以前捐助过的小伙伴请给我发个邮件（lucien@lucien.ink）告知一下自己的 `ID` ，我会加到列表里，之前的收款码看不了付款人 `ID` 的全称。

##### 2019-05-06 补充

然而过去了这么久之前匿名捐的小伙伴们没几个私聊我的 QAQ，所以就直接在这里把支付宝、微信里显示给我的 ID 列出来吧，特意附上了时间，欢迎各位小伙伴来认捐，谢谢各位小伙伴的支持 Orz。

| ID | 金额 | 时间 |
|--|--|--|
| [@Irene Liu](https://github.com/ireneliuqaq) | 1.98 | 2018-08-24 |
| \*晨晨 | 0.99 | 2018-08-28 |
| \*绍延 | 0.99 | 2018-09-04 |
| \*鱼 | 0.99 | 2018-09-04 |
| d\*d | 0.99 | 2018-09-06 |
| \*特 | 0.99 | 2018-09-30 |
| L\*L | 0.99 | 2018-11-28 |
| \*海燕 | 0.99 | 2018-12-05 |
| \*乐 | 1.98 | 2018-12-10 |
| \*夏 | 0.99 | 2018-12-11 |
| a\*X | 0.99 | 2018-12-15 |
| \*书轩 | 0.99 | 2018-12-23 |
| \*磊毅 | 1.98 | 2018-12-27 |
| Rajy | 6.66 | 2019-01-05 |
| Jim Zhou | 0.88 | 2019-04-12 |
| LXF | 2.22 | 2019-02-02 |
| \*政 | 0.99 | 2019-02-17 |
| [@Edwiv](https://github.com/Edwiv) | 1.99 | 2019-03-02 |
| [@louyu666](https://github.com/louyu666) | 2.33 | 2019-04-18 |
| [@oierwanhong](https://github.com/oierwanhong) | 9.99 | 2019-05-02 |
| [@Albertliuzw](https://github.com/Albertliuzw) | 1.50 | 2019-05-17 |
| [@EndangeredFish](https://github.com/EndangeredF1sh) | 9.90 | 2019-06-05 |
| \*舟 | 1.00 | 2019-06-11 |
| [@异或和](https://github.com/XorSum) | 6.60 | 2019-06-17 |
| \*伟 | 2.00 | 2019-06-18 |
| \*楠 | 10.00 | 2019-06-27 |
| \*. | 1.00 | 2019-07-03 |
| \*栈 | 100.00 | 2019-07-06 |
| \*珊珊 | 1.50 | 2019-07-13 |
| n\*l | 0.88 | 2019-07-14 |
| Rhythm~ | 1.99 | 2019-08-14 |
| \* 天翔 | 2.33 | 2019-08-28 |
| \*景周 | 6.60 | 2019-09-09 |
| 3402 | 5.00 | 2019-09-09 |
| \*菡 | 10.00 | 2019-09-23 |
| \*瑶 | 23.33 | 2019-10-14 |
| \*晨 | 0.99 | 2019-11-08 |
| 螃蟹 | 6.66 | 2019-11-13 |
| \*库 | 2.33 | 2019-11-21 |
| \*橙 | 0.10 | 2019-12-21 |
| [@NamelessAshone](https://github.com/NamelessAshone) | 20.00 | 2020-02-20 |
| \*超 | 6.00 | 2020-02-28 |
| [@Aphroq](https://github.com/Aphroq) | 6.66 | 2020-03-30 |

#### 谢谢老板

![谢谢老板](https://github.com/LucienShui/gitcdn/blob/master/pasteme_donate.png?raw=true)

### Browsers support

Modern browsers and Internet Explorer 10+.

## Browsers support

| [<img src="https://raw.githubusercontent.com/alrra/browser-logos/master/src/edge/edge_48x48.png" alt="IE / Edge" width="24px" height="24px" />](http://godban.github.io/browsers-support-badges/)<br>IE / Edge | [<img src="https://raw.githubusercontent.com/alrra/browser-logos/master/src/firefox/firefox_48x48.png" alt="Firefox" width="24px" height="24px" />](http://godban.github.io/browsers-support-badges/)<br>Firefox | [<img src="https://raw.githubusercontent.com/alrra/browser-logos/master/src/chrome/chrome_48x48.png" alt="Chrome" width="24px" height="24px" />](http://godban.github.io/browsers-support-badges/)<br>Chrome | [<img src="https://raw.githubusercontent.com/alrra/browser-logos/master/src/safari/safari_48x48.png" alt="Safari" width="24px" height="24px" />](http://godban.github.io/browsers-support-badges/)<br>Safari | [<img src="https://raw.githubusercontent.com/alrra/browser-logos/master/src/safari-ios/safari-ios_48x48.png" alt="iOS Safari" width="24px" height="24px" />](http://godban.github.io/browsers-support-badges/)<br>iOS Safari | [<img src="https://raw.githubusercontent.com/alrra/browser-logos/master/src/samsung-internet/samsung-internet_48x48.png" alt="Samsung" width="24px" height="24px" />](http://godban.github.io/browsers-support-badges/)<br>Samsung | [<img src="https://raw.githubusercontent.com/alrra/browser-logos/master/src/opera/opera_48x48.png" alt="Opera" width="24px" height="24px" />](http://godban.github.io/browsers-support-badges/)<br>Opera |
| --------- | --------- | --------- | --------- | --------- | --------- | --------- |
| IE11, Edge| last 2 versions| last 2 versions| last 2 versions| last 2 versions| last 2 versions| last 2 versions

## 版权所有

Copyright &copy; 2017-2019 [Lucien Shui](http://www.lucien.ink) All Rights Reserved

## 特别鸣谢

> 按账号的字典序排序

### 前端

+ [@whitelord](https://github.com/alexischiang)
+ [@Ryan Lee](https://github.com/ryanlee2014)
+ [@fishtwlve](https://github.com/yjy2019)

### 后端

#### Golang

+ [@bilibili2834451280](https://github.com/bilibili2834451280)

#### Java

+ [@moyu Programmer](https://github.com/1059438020)
+ [@Acerkoo](https://github.com/Acerkoo)
+ [@Irene Liu](https://github.com/ireneliuqaq)
+ [@Ryan Lee](https://github.com/ryanlee2014)
+ [@UPCPHY](https://github.com/UPCPHY)
+ [@V-ioi-V](https://github.com/V-ioi-V)
+ [@Runn](https://github.com/XRZSJHHK)

## 免责声明

本平台只提供文本分享的载体，与所有文本内容均没有任何联系。



CHANGELOG.md
# Change Log

### 2019-05-01

1. Remove travis badge in `README.md`
2. Update Document
3. Update donate list

## v3.1.0F-3.1.0B

### 2019-10-10

1. Add badge in `README.md`

### 2019-07-24

1. Add `github.com/gin-contrib/cors`

### 2019-07-22

1. Add beat

## v3.1.0F-3.0.1B

### 2019-09-02

1. Using CDN for static file

## v3.0.3F-3.0.1B

### 2019-08-25

1. Update doc
2. Add webhook
3. Update donate list

### 2019-07-24

1. Update `docker-compose.yml`

### 2019-07-23

1. Change default deploy to docker deploy
2. Remove a comment from `docker-compose.yml`

## Version 3.0.1

### 2019-07-12

1. Add [DOCKER_DEPLOY.md](./doc/DOCKER_DEPLOY.md)
2. Add [.travis.yml](./.travis.yml) again
3. Update `docker-compose.yml`, add healthcheck
4. Update `DEPLOY.md`, add docker deploy statement

### 2019-07-08

1. Update `DEPLOY.md`, add `required`
2. Update `README.md`

api.php
<?php
/**
 * Created by Lucien
 * Date: 2018/8/26
 * Time: 11:03
 */
require_once('lib/tableEditor.php');
if (isset($_GET['token'])) {
    $token = $_GET['token'];
    $key = '';
    $passwd = '';
    $pos = strpos($token, ",");
    if ($pos === false) {
        $key = $token;
    } else {
        $key = substr($token, 0, $pos);
        $passwd = substr($token, $pos + 1, strlen($token) - $pos + 1);
    }
    $len = strlen($key);
    $chFlag = false;
    for ($i = 0; $i < $len; $i++) {
        if ($key[$i] < '0' || $key[$i] > '9') {
            $chFlag = true;
            break;
        }
    }
    $table = new tableEditor();
    $array = $table->query($key);
    if ($array === false) {
        echo json_encode(array(
            'code' => 404,
            'data' => array(
                'message' => 'key not found',
            )
        ));
    }
    else {
        $flag = false;
        $sys_passwd = $array['passwd'];
        if (!empty($sys_passwd)) {
            if (!empty($sys_passwd) && md5($passwd) == $sys_passwd) $flag = true;
        } else {
            $flag = true;
        }
        if ($flag) {
            if (isset($_GET['browser'])) {
                echo json_encode(array(
                    'code' => 200,
                    'data' => array(
                        'message' => 'success',
                        'content' => $array['text'],
                        'type' => $array['type'],
                    )
                ));
            } else {
                echo str_replace("\r", "", htmlspecialchars_decode($array['text'])) . "\n";
            }
            if ($chFlag) {
                $table->erase($key);
            }
        } else {
            echo json_encode(array(
                'code' => 406,
                'data' => array(
                    'message' => 'passwd wrong',
                )
            ));
        }
    }
} else {
    echo json_encode(array(
        'code' => 500,
        'data' => array(
            'message' => 'wrong args',
        )
    ));
}
